Lib/idlelib/calltip.py

```diff
@@ -55,18 +55,46 @@ def refresh_calltip_event(self, event):
             self.open_calltip(False)
 
     def open_calltip(self, evalfuncs):
-        self.remove_calltip_window()
-
         hp = HyperParser(self.editwin, "insert")
         sur_paren = hp.get_surrounding_brackets('(')
         if not sur_paren:
+            # Not inside parentheses: Don't open a calltip, and close one if
+            # currently open.
+            self.remove_calltip_window()
             return
+
+        if self.active_calltip:
+            opener_line, opener_col = map(int, sur_paren[0].split('.'))
+            if (
+                (opener_line, opener_col) ==
+                (self.active_calltip.parenline, self.active_calltip.parencol)
```

Comment:
I presume this solves most if not all of the flashing.

Reply:
Indeed, but only combined with not calling `remove_calltip_window()` at the top of this function to begin with.